components/universal-widget/src/UniversalWidget.jsx

```diff
@@ -136,10 +136,12 @@ export default class UniversalWidget extends Widget {
     }
 
     handleWidgetData(data) {
-        this.setState({
-            metadata: data.metadata ? data.metadata : this.state.metadata,
-            data: data.data
-        })
+        if(data.data.length != 0){
```

Comment:
Can `data.data` be `null`? If so this will result a NPE.
Shall we improve this to `data.data && data.data.length`?

Reply:
data.data cannot be null. Its an array list.[1]
[1] https://github.com/wso2/carbon-analytics/blob/master/components/org.wso2.carbon.data.provider/src/main/java/org/wso2/carbon/data/provider/rdbms/RDBMSStreamingDataProvider.java#L65